src/main/java/com/mongodb/spark/sql/connector/config/Configurable.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.mongodb.spark.sql.connector.config;
+
+/** Represents a configurable class */
+public interface Configurable {
+
+  /**
+   * Configure this class with the {@link MongoConfig}.
+   *
+   * @param config the current configuration
+   */
+  void configure(MongoConfig config);
```

Comment:
So `Configurable.configure` is similar to a method annotated with [`@PostConstruct`](https://docs.oracle.com/javase/8/docs/api/javax/annotation/PostConstruct.html) (while it is part of the Java SE 8 API, it was later moved to [Jakarta Annotations](https://jakarta.ee/specifications/annotations/2.0/apidocs/jakarta.annotation/jakarta/annotation/postconstruct)).

> Users can provide names of classes
> allows users to pass configurable instances

In the current PR I see only the former, where a class name may be specified and mongo-spark creates its instance and calls `Configurable.configure` on it. Was the latter a typo, or are we actually expecting to have an API that allows users to pass `Configurable` objects (not names of classes) to mongo-spark, where we will inject a configuration by calling `configure`?

The above is an important question, because if it is always the case that mongo-spark creates an instance of `Configurable` and calls `configure` before using the instance, then the `Configurable` interface only complicates matters without providing any benefit. In this case we can instead say that the classes users specify must have either a `public` constructor that takes no parameters, or a `public` constructor that takes a single parameter of type `MongoConfig`.